lib/index.js

```diff
@@ -1 +1,98 @@
-module.exports = require('./binding/node_valhalla.node');
+var valhalla = require('./binding/node_valhalla.node');
+const genericPool = require("generic-pool");
+
+module.exports = function(configStr, minWorkers, maxWorkers) {
+    var Valhalla = valhalla(configStr);
+
+    function createWorker() {
+        return new Promise(function(resolve, reject) {
+            var valh;
+            try {
+                valh = new Valhalla(configStr);
+                resolve(valh);
+            } catch (e) {
+                reject(e);
+            }
+        });
+    }
+
+    const actorFactory = {
+        create: createWorker,
+        destroy: function(actor) {
+            return new Promise(function(resolve, reject) {
+                try {
+                    delete actor;
+                    resolve();
+                } catch (e) {
+                    reject(e);
+                }
+            });
+        }
+    };
+
+    const opts = {
+        max: maxWorkers || 1,
+        min: minWorkers || 1
+    };
+    
+    return function Actor(configStr) {
+        try {
+            new Valhalla(configStr);
+        } catch (e) {
+            throw e;
+        }
+
+        const actorPool = genericPool.createPool(actorFactory, opts);
+        function actorMethod(methodName, request, cb) {
```

Comment:
Optional refactor: `actorMethod` can be a function that takes `(methodName)` and returns a function taking `(request, cb)`, which simplifies setting methods on `this`. I think this should work:

```
function actorMethodFactory = (methodName) => {
  return function(request, cb) {
    ...
  }
}

this.route = actorMethodFactory('route');
this.locate = actorMethodFactory('locate');
...
```